goal_src/engine/game/generic-obs.gc

```diff
@@ -136,6 +127,100 @@
   (none)
   )
 
+(defstate swingpole-stance (swingpole)
+  :code
+  (behavior ()
+   (while #t
+    (when
+     (and
+      *target*
+      (<
+       (vector-vector-distance
+        (-> self root trans)
+        (the-as vector (&-> (-> *target* control) unknown-cspace00 bone))
+        )
+       (-> self range)
+       )
+      (logtest? (-> *target* control root-prim prim-core action) 64)
+      (<
+       (the-as float (-> *target* control unknown-cspace00 param0))
+       (+ (-> self root trans y) (* 0.5 (-> self range)))
+       )
+      )
+     (let ((a1-1 (new 'stack-no-clear 'event-message-block)))
+      (set! (-> a1-1 from) self)
+      (set! (-> a1-1 num-params) 1)
+      (set! (-> a1-1 message) 'pole-grab)
+      (set! (-> a1-1 param 0) (the-as uint self))
+      (if (send-event-function *target* a1-1)
+       (go swingpole-active)
+       )
+      )
+     )
+    (suspend)
+    )
+   (none)
+   )
+  )
+
+(defstate swingpole-active (swingpole)
+  :code
+  (behavior ()
+   (suspend)
+   (while
+    (and
+     *target*
+     (= (handle->process (-> *target* control unknown-handle10)) self)
+     )
+    (suspend)
+    )
+   (let ((gp-0 (-> *display* base-frame-counter)))
+    (until (>= (- (-> *display* base-frame-counter) gp-0) 150)
+     (suspend)
+     )
+    )
+   (go swingpole-stance)
+   (none)
+   )
+  )
+
+(defmethod copy-defaults! swingpole ((obj swingpole) (arg0 res-lump))
+  "Copy defaults from the entity."
+  (stack-size-set! (-> obj main-thread) 128)
+  (logior! (-> obj mask) (process-mask actor-pause))
+  (set! (-> obj root) (new 'process 'trsq))
+  (set! (-> obj root trans quad) (-> arg0 extra trans quad))
+  (quaternion-copy! (-> obj root quat) (-> (the-as entity-actor arg0) quat))
+  (vector-identity! (-> obj root scale))
+  (vector-y-quaternion! (-> obj dir) (-> obj root quat))
+  (set! (-> obj dir y) 0.0)
+  (vector-normalize! (-> obj dir) 1.0)
+  (set! (-> obj range) 12288.0)
+  (set! (-> obj edge-length) 8192.0)
+  (go (the-as (state none) swingpole-stance))
```

Comment:
is this really what the decompiler comes up with here? Something doesn't look right, maybe you still have a cast on this?